### PR TITLE
dev-python/kaitaistruct: Introducing new package

### PR DESCRIPTION
**Pull request by bucefal91**
Introducing dev-python/kaitaistruct into Gentoo repository.

Closes: https://bugs.gentoo.org/654860
Package-Manager: Portage-2.3.33, Repoman-2.3.9

### CODE REVIEW

File: dev-python/kaitaistruct/kaitaistruct-0.7.ebuild [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 python3_{3,4,5} )
+
+inherit distutils-r1
+
+DESCRIPTION="Kaitai Struct runtime for Python"
+HOMEPAGE="http://kaitai.io/"
```
Comment (ConiKost) [inline dev-python/kaitaistruct/kaitaistruct-0.7.ebuild]: HTTPS please

File: dev-python/kaitaistruct/kaitaistruct-0.7.ebuild [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
```
Comment (ConiKost) [inline dev-python/kaitaistruct/kaitaistruct-0.7.ebuild]: Drop it, if empty.

**Comment by bucefal91** [comment]
@ConiKost Hello :) thanks for reviewing my stuff!

I believe I have processed your feedback (will be processing now in other pull requests too).

File: dev-python/kaitaistruct/Manifest [hunk 3]
```diff
@@ -0,0 +1 @@
+DIST kaitaistruct-0.7.tar.gz 5117 BLAKE2B d7ef62414ca07c48f8a813925d0ab157bb1c198619795b6aa99de1c9415eea397e3b24ee2f28bbc1f4315ec7cb0b81fbe7306c1444104365cd4a455ce4e54d82 SHA512 1ee2e5d1e1de6a85dcaea684bee639f6b08bc0f06b8357fe388c339287388136638d11d301319a344364dc9cf98d3247fb6fc2d8332d9aaee6ee828e5650b447
```
Comment (ConiKost) [inline dev-python/kaitaistruct/Manifest]: Why not directly newest 0.8 release?

File: dev-python/kaitaistruct/kaitaistruct-0.7.ebuild [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 python3_{3,4,5} )
```
Comment (ConiKost) [inline dev-python/kaitaistruct/kaitaistruct-0.7.ebuild]: python3_6 is missing, is this correct?

**Comment by bucefal91** [comment]
Somehow my search engine (duckduck go) threw me onto 0.7 and I didn't bother to make sure it was the latest release (I just assumed I would magically land onto the freshest one from the search results).

So I added 0.8.ebuild too. I also looked into supported versions as reported by https://pypi.org/project/kaitaistruct/0.{7,8}/ and written corresponding stuff into PYTHON_COMPAT. I also removed python3_3 as per https://wiki.gentoo.org/wiki/Project:Python/Implementations it's no longer supported.

File: dev-python/kaitaistruct/kaitaistruct-0.7.ebuild [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# Copyright 1999-2018 Gentoo Foundation
```
Comment (mgorny) [inline dev-python/kaitaistruct/kaitaistruct-0.7.ebuild]: Remove the old ebuild.

File: dev-python/kaitaistruct/kaitaistruct-0.8.ebuild [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+inherit distutils-r1
+
+DESCRIPTION="Kaitai Struct runtime for Python"
+HOMEPAGE="https://kaitai.io/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
```
Comment (mgorny) [inline dev-python/kaitaistruct/kaitaistruct-0.8.ebuild]: Missing dep on setuptools.

I suppose running tests is not an option here.

Comment (bucefal91) [reply dev-python/kaitaistruct/kaitaistruct-0.8.ebuild]: Hello @mgorny 

I've removed the older ebuild; added the dependency on setuptools. As far as I understand https://github.com/kaitai-io/kaitai_struct the "tests" folder is another git repository and `ebuild kaitaistruct-0.8.ebuild fetch` fetches a tarball without it.

I also squashed all my commits into a single one.

**Comment by mgorny** [comment]
Is this also part of something bigger? If yes, then I would propose I'll just mark this one as reviewed but merge them all together.

**Comment by bucefal91** [comment]
@mgorny , yes, it is a dependency of net-proxy/mitmproxy-3.0.4 https://bugs.gentoo.org/572170

I might be wrong, but I think repoman wouldn't allow me to commit or QA bot would throw tons of errors if I file a pull request for mitmproxy and one of its dependencies would be missing in the main gentoo tree (though present in my overlay). I have a ready & functional mitmproxy-3.0.4.ebuild; I didn't file pull request for it precisely because the dependencies were missing.

**Comment by mgorny** [comment]
Let me rephrase. Let's review all the deps you have filed already, then close them and file one pull request containing all packages. Would that work for you?

**Comment by bucefal91** [comment]
Yes, sure :) 

Currently open I have for net-proxy/mitmproxy:
* this one
* https://github.com/gentoo/gentoo/pull/8262 (I have things to do there... will try to process your feedback today/tomorrow)
* https://github.com/gentoo/gentoo/pull/8261 (also I have to go through your feedback there)

And for authenticator:
* https://github.com/gentoo/gentoo/pull/8282 (seems like nobody from proxying maintainers have reviewed it yet). At least I haven't got there not a single comment nor positive or negative.

**Comment by bucefal91** [comment]
@mgorny In your last comment to this pull request you said you'd like to merge a bunch of dev-python/* packages I've been working on once they are all settled. Looks like they are all reviewed (some of them are already merged by you), the ones that are still not:
* this one.
* https://github.com/gentoo/gentoo/pull/8262 - I've gone through your feedback and I hope it's ready for merge too.

Do you want me to squash it into a single commit or how do you want to proceed? :)
